### ISSUE TITLE
Cannot access sails config in grunt task

### ISSUE THREAD

**Issue by vollnhals** [balderdashy/sails#2560]
I have written a custom grunt task that needs to access the sails config.

```
module.exports = function (grunt) {
  grunt.registerTask('customtask', function () {
    console.log(sails.config.session.secret);
  });
};
```

But i cannot access the sails object. I get the following error:

```
Warning: sails is not defined Use --force to continue.

Aborted due to warnings.
```

How can I access the sails config? 
(Also how can I access the services?)

**Comment by andrewcarreiro** [comment]
I believe this is due to Grunt running before Sails does. What I've done in the past is just manually require the config file:

`var configLocal = require('./config/local.js');`

**Comment by elennaro** [comment]
Is there a better solution?
I like that Sails builds all the assets each run, and it depends on the environment I set.
And I like that Sails has a powerful config that also depends on the environment.
It'll be good to unite this two for custom grunt tasks.

**Comment by jvanalst** [comment]
Curses. This is what I feared. Now I'll have to do something fairly hacky.

I need to access the sails base url because I'm running behind a proxy (in a subfolder), which breaks all the absolute paths that are in the default sails-linker.

**Comment by sailsbot** [comment]
Thanks for posting, @vollnhals.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
